src/App.js

```diff
@@ -0,0 +1,35 @@
+import React, { useState } from 'react';
+import { Route, Routes } from 'react-router-dom';
+import './App.css';
+import Login from './components/login/login';
+import NotFound from './components/pagenotfound.js/notfound';
+import Furniture from './components/dashboard/furniture/Furniture';
+import Home from './components/dashboard/home';
+import Signup from './components/login/signup';
+import Dashboard from './components/dashboard/dashboard';
+import FurnitureForm from './components/dashboard/furniture/FurnitureForm';
+import AppointmentForm from './components/dashboard/appointment/appointmentForm';
+import AppointmentsList from './components/dashboard/appointment/appointmentsList';
+import DeleteFurniture from './components/dashboard/furniture/DeleteFurniture';
+
+function App() {
```

Comment:
- [ ] When I create and login as a new user, the localStorage data is crated correctly:

![image](https://github.com/anbehindY/furniture-mall-frontend/assets/57421823/3d56af50-0815-4b81-94ff-b235b1a9413f)

However, the page is blank:

![image](https://github.com/anbehindY/furniture-mall-frontend/assets/57421823/6fc5ce4b-31b1-449f-ac5d-42894048ceb8)

This is because the request for furniture to the server has failed:

![image](https://github.com/anbehindY/furniture-mall-frontend/assets/57421823/3892ba1b-6f93-4486-9a6e-33dd4a2e5ecf)

![image](https://github.com/anbehindY/furniture-mall-frontend/assets/57421823/9795bfad-66bf-4257-b802-a93b8c1bfacd)

## Explanation

As a _**NEW**_ user, this fails on first login because it makes the request to the server for furniture immediately, before the token and the user have been populated on the local storage. 

![image](https://github.com/anbehindY/furniture-mall-frontend/assets/57421823/4b1d5573-6398-4678-9590-c4d5308699b0)

Thus, the operation is unauthorized because the token in the local storage does not exist yet.

I need to reload the page for this to work:

### After reloading:

![image](https://github.com/anbehindY/furniture-mall-frontend/assets/57421823/abb0890e-c234-4a12-868e-816fc8b564e2)



This won't happen on your end because your local storage already exists.  In order to replicate this bug, kindly remove local storage or use an incognito tab.

Kindly, make it so the request is successful on the first try, without having to reload the page as a new user.  

_tip:  You could use `async ` for this so that on login the local storage is created first, and THEN, the server makes the request for furnitore._

Reply:
Fixed. Thanks you.